vscode/src/webviewPanel.ts

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  QscEventTarget,
+  VSDiagnostic,
+  getCompilerWorker,
+  log,
+} from "qsharp-lang";
+import {
+  commands,
+  ExtensionContext,
+  Uri,
+  ViewColumn,
+  Webview,
+  WebviewPanel,
+  window,
+} from "vscode";
+import { isQsharpDocument } from "./common";
+import { reSampleData } from "./reSampleData";
+
+const histogramRunTimeoutMs = 1000 * 60 * 5; // 5 minutes
+
+export function registerWebViewCommands(context: ExtensionContext) {
+  QSharpWebViewPanel.extensionUri = context.extensionUri;
+
+  const compilerWorkerScriptPath = Uri.joinPath(
+    context.extensionUri,
+    "./out/compilerWorker.js",
+  ).toString();
+
+  // Stub for now to prototype the UX
+  // Add the following to settings.json to enable the command:
+  //   "Q#.experimental-re": true
+  context.subscriptions.push(
+    commands.registerCommand("qsharp-vscode.showRe", async () => {
+      const message = {
+        command: "estimate",
+        estimatesData: reSampleData,
+      };
+      sendMessageToPanel("estimate", true, message);
+    }),
+  );
+
+  context.subscriptions.push(
+    commands.registerCommand("qsharp-vscode.showHelp", async () => {
+      const message = {
+        command: "help",
+      };
+      sendMessageToPanel("help", true, message);
+    }),
+  );
+
+  context.subscriptions.push(
+    commands.registerCommand("qsharp-vscode.showHistogram", async () => {
```

Comment:
This spurred some UX confusion in the Monday demo, and I tend to agree that it's a bit counterintuitive that "show histogram" does real work (i.e. runs the program) besides just showing a window. I think a more cohesive UX here would be to use the existing "Run Q# File" command to actually run the program, and then display the Histogram based on the simulation that was just run. It just feels strange that we have two ways of running a Q# program now, and one is called "show" and shows a histogram, and the other one doesn't. How would a user know which one to pick?

Perhaps for now, rename to something like "Run Q# File (Histogram)"? I'm not sure if I like my own suggestion.

Not for this PR, but I think taking a holistic look at the UX and maybe aligning some of the disjoint experiences would be nice before we ship.

Reply:
Yeah. The existing 'Run' (Ctrl+F5) experience does what you'd expect, and runs the code once, outputting messages to the debug console. For some algorithms (or when developing) that's exactly what you want, and histograms aren't interesting (e.g. all the samples on the playground that default to 1 shot are that way because a histogram is of no value).

When you run a histogram you typically are interested in running hundreds (or thousands) of shots and seeing the statistical results, and the thousands of lines that likely end up in your debug console are not interesting.

I do see them as two very different use cases, but we should agree on how to make the experience intuitive.

Reply:
I changed the command text to 'Run file and show histogram' (it's already prefixed with 'Q#')